src/test/groovy/graphql/GraphQLTest.groovy

```diff
@@ -39,6 +41,36 @@ class GraphQLTest extends Specification {
 
     }
 
+    def "incomplete builder overrides"() {
+        given:
+        DataFetcher dataFetcher = new DataFetcher() {
+            @Override
+            Object get(DataFetchingEnvironment environment) {
+                return environment.arguments.entrySet().getAt(0).value;
+            }
+        }
+
+        GraphQLObjectType objectType = newObject()
+                .name("root")
+                .field(newFieldDefinition()
+                    .type(GraphQLString)
+                    .name("field")
+                    .argument(newArgument()
+                        .name("arg")
+                        .type(GraphQLString))
+                    .dataFetcher(dataFetcher))
+        .build()
+
+        GraphQLSchema schema = newSchema()
+                .query(objectType).build()
```

Comment:
Could we add an override to `query()` that accepts a `GraphQLObjectType.Builder`, so that we don't have to call `.build()` above? Same with `mutation()` then.

Can we do the same with `GraphQLFieldDefinition#type()`? For that it would be an overload that accepts a `GraphQLObjectType.Builder`. I think thats the only builder it should accept.